tests/e2e/transaction/signTransaction.test.ts

```diff
@@ -216,3 +218,46 @@ describe("sign transaction", () => {
     });
   });
 });
+
+test.only("test", async () => {
+  const fromAccount = Account.generate();
+  const newAccount = Account.generate();
+
+  await aptos.fundAccount({ accountAddress: fromAccount.accountAddress, amount: 1_000_000_000 });
+  await aptos.fundAccount({ accountAddress: newAccount.accountAddress, amount: 1_000_000_000 });
+
+  const accountInfo = await aptos.getAccountInfo({
+    accountAddress: fromAccount.accountAddress,
+  });
+
+  const challenge = await aptos.createProofChallenge({
+    struct: "0x1::account::RotationProofChallenge",
+    data: [
+      BigInt(accountInfo.sequence_number),
+      fromAccount.accountAddress,
+      accountInfo.authentication_key,
+      newAccount.publicKey.toUint8Array(),
+    ],
+  });
+
+  const proofSignedByCurrentPrivateKey = aptos.signProofChallenge({ challenge, signer: fromAccount });
+  const proofSignedByNewPrivateKey = aptos.signProofChallenge({ challenge, signer: newAccount });
+
+  const transaction = await aptos.transaction.build.simple({
+    sender: fromAccount.accountAddress,
+    data: {
+      function: "0x1::account::rotate_authentication_key",
+      functionArguments: [
+        new U8(fromAccount.signingScheme), // from scheme
+        MoveVector.U8(fromAccount.publicKey.toUint8Array()),
+        new U8(newAccount.signingScheme), // to scheme
+        MoveVector.U8(newAccount.publicKey.toUint8Array()),
+        MoveVector.U8(proofSignedByCurrentPrivateKey.toUint8Array()),
+        MoveVector.U8(proofSignedByNewPrivateKey.toUint8Array()),
+      ],
+    },
+  });
+
+  const response = await aptos.signAndSubmitTransaction({ signer: fromAccount, transaction });
+  console.log("response", response);
```

Comment:
This looks good to me, but would be nice to test the whole flow (ie. 1. Create Prroof. 2. Create transaction with proof. 3. submit transaction and make sure it goes through)

The reason for this is I've had a lot of issues in the past actually getting the challenge in the right format. And it also seems really easy to break. Everything looks fine in the PR for contructing the challenges but...

Reply:
The test here 
- creates a proof
- signs the proof
- create transaction with proof
- submits successfully to chain

Were you looking for a different flow example?